### DISCUSSION TITLE
Implement Secret Sauce AI's wake word detector

### DISCUSSION THREAD

**Discussion by nerdaxic** [nerdaxic/glados-voice-assistant#6] [Roadmap]
1. Make robot listen to "hey glados" using machine learning model 
2. Remove pocketsphinx from implementation

_Read more: [Secret Sauce AI's wakeword project](https://github.com/secretsauceai/secret_sauce_ai/wiki/Wakeword-Project)_

**Comment by nerdaxic** [comment]
@salleq, @eternalliving and @AmateurAcademic could you guys: 

1. Collect your voice sample set with this tool 
2. Zip the files and with the folder structure
3. Share to me in a zip file via google drive, wetransfer etc. Can send privately so your voice wont be published.
 
I can train the wakework detector model to be a bit more general.
https://github.com/secretsauceai/wakeword-data-collector

**Here are the specs as needed:**

**Trigger:** "hey glados"
**Syllables used:** "hey, gla, dos"
**Name:** Use your github nick as the name when tool requests it. 

Tool is also under developement so any notes you take during usage is much appreciated!

**Reply by salleq** [reply]
Hey, for sure! I'll book some time for this.

**Reply by eternalliving** [reply]
Yea I can work on that... I have an Aussie accent so hopefully that doesn't mess with your model too much ;)
Looking forward to being able to use that wakeword though... "hey alexa" just seems wrong... Glados should have some comment that we're calling her the wrong name :P

**Reply by nerdaxic** [reply]
Thanks guys. @eternalliving aussie accent not bad mate! Idea is to have variety.
You can already change the "hey alexa" into whatever in the settings file already, it is not very good tho. I'm not planning to make different trigger-word detection functions for google and alexa trigger words just for lolz. We have science to do!

**Reply by AmateurAcademic** [reply]
If you are collecting for several people: 

the users don't have to do a full collection after at least one full collection. I would recommend just doing one wakeword collection (16 samples of the wakeword), then either one TV or conversation collection (which is more to get the dynamics with the mic and the room), the syllables and sequential parts can be helpful though to generally collect (ie for 'hey glados': 'hey', 'gla' 'dos', 'hey gla', 'glados').

But it is still important to stress: make sure to do the collection on the mic you will actually use, in the place you will use it. 

It would be really great if people would write what mic they are using as a note.

I will use the pseye mic in my livingroom and will note that.

If I can help or support otherwise, let me know.

**Reply by salleq** [reply]
@nerdaxic sent you a dm in twitter

**Reply by eternalliving** [reply]
@nerdaxic Am not sure how to send privately or give you a private link.

**Reply by nerdaxic** [reply]
@salleq thanks for the data, got it!

@eternalliving you can send me link with this, unless you are a robot
https://www.henrirantanen.fi/contact/

@AmateurAcademic big thanks, appreciate the instructions and help!

**Comment by nerdaxic** [comment]
[Mycroft Precise](https://github.com/MycroftAI/mycroft-precise) wake-word engine apparently requires specific ye-olde version of TensorFlow, which requires python 3.7. Preliminary "hey glados" model has been already trained and uploaded to repo.

I will implement this after installing everything into python virtual environment first, so we can have all the right dependencies attached.

**Reply by salleq** [reply]
I don't know if it's of any use, but my docker stuff has exactly Python 3.7.
https://github.com/salleq/glados-docker

**Reply by nerdaxic** [reply]
I have chatted with a few developers about using docker for this and so far decided against it. It is usually difficult to get the audio working in and out from the container on different platforms. 

Might work for you and I'm GLaD if it does, but people have ran into issues with it. Also not sure if it can use GPU the same way - afterall there is machine-vision stuff on the roadmap and I'm tired of fighting even before the battle 🗡️ 

Nothing stops us from shoving this into docker later with the venv! Im open to ideas but this was my reasoning behind not using docker so far!

**Reply by salleq** [reply]
Yeah I probably misunderstood you - I didn't mean for you to migrate the whole project to docker, just that if it helped you test anything.
I definitely agree with you, just keep on working on VENV and I keep using it through docker as long as it works :)

**Reply by nerdaxic** [reply]
I was going to shove it into docker before I ran into audio issues 🥲

**Reply by salleq** [reply]
One thing that didn't work was the python script for playing audio, that's why I asked if we can change it to aplay. But of course there could be other issues as well.
